### ISSUE TITLE
OMV 5 : Can't apply iptable rules.

### ISSUE THREAD

**Issue by luluwebmaster** [openmediavault/openmediavault#374]
Hello,

Recently, I have installed OMV 5 ( Beta ), but now, I try to make iptables rules, and when I apply modifications, I have this error : 

`Failed to execute command 'export PATH=/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin; export LANG=C.UTF-8; omv-salt deploy run iptables 2>&1' with exit code '1': /usr/lib/python3/dist-packages/salt/modules/file.py:32: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working from collections import Iterable, Mapping debian: ---------- ID: configure_ifupdown_iptables_rules Function: file.managed Name: /etc/network/if-pre-up.d/openmediavault-iptables Result: True Comment: File /etc/network/if-pre-up.d/openmediavault-iptables updated Started: 15:33:55.487194 Duration: 60.386 ms Changes: ---------- diff: --- +++ @@ -21,5 +21,5 @@ iptables -A INPUT -p udp --source 255.255.255.0/24 --destination 192.168.1.5 --dport 445 -j ACCEPT iptables -A INPUT -p udp --source 255.255.255.0/24 --destination 192.168.1.5 --dport 137:138 -j ACCEPT iptables -A INPUT -p tcp --source 255.255.255.0/24 --destination 192.168.1.5 --dport 22 -j ACCEPT -iptables -A INPUT -p all -j REJECT +iptables -A INPUT -p all -j ACCEPT ---------- ID: apply_ifupdown_iptables_rules Function: cmd.run Name: /etc/network/if-pre-up.d/openmediavault-iptables Result: False Comment: Command "/etc/network/if-pre-up.d/openmediavault-iptables" run Started: 15:33:55.548289 Duration: 11.512 ms Changes: ---------- pid: 3395 retcode: 1 stderr: trap: 15iptables: bad trap stdout: Summary for debian ------------ Succeeded: 1 (changed=2) Failed: 1 ------------ Total states run: 2 Total run time: 71.898 ms`

Full details : 

```
Erreur #0:
OMV\ExecException: Failed to execute command 'export PATH=/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin; export LANG=C.UTF-8; omv-salt deploy run iptables 2>&1' with exit code '1': /usr/lib/python3/dist-packages/salt/modules/file.py:32: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
  from collections import Iterable, Mapping
debian:
----------
          ID: configure_ifupdown_iptables_rules
    Function: file.managed
        Name: /etc/network/if-pre-up.d/openmediavault-iptables
      Result: True
     Comment: File /etc/network/if-pre-up.d/openmediavault-iptables updated
     Started: 15:33:55.487194
    Duration: 60.386 ms
     Changes:
              ----------
              diff:
                  ---
                  +++
                  @@ -21,5 +21,5 @@
                   iptables -A INPUT -p udp --source 255.255.255.0/24 --destination 192.168.1.5 --dport 445 -j ACCEPT
                   iptables -A INPUT -p udp --source 255.255.255.0/24 --destination 192.168.1.5 --dport 137:138 -j ACCEPT
                   iptables -A INPUT -p tcp --source 255.255.255.0/24 --destination 192.168.1.5 --dport 22 -j ACCEPT
                  -iptables -A INPUT -p all -j REJECT
                  +iptables -A INPUT -p all -j ACCEPT
----------
          ID: apply_ifupdown_iptables_rules
    Function: cmd.run
        Name: /etc/network/if-pre-up.d/openmediavault-iptables
      Result: False
     Comment: Command "/etc/network/if-pre-up.d/openmediavault-iptables" run
     Started: 15:33:55.548289
    Duration: 11.512 ms
     Changes:
              ----------
              pid:
                  3395
              retcode:
                  1
              stderr:
                  trap: 15iptables: bad trap
              stdout:

Summary for debian
------------
Succeeded: 1 (changed=2)
Failed:    1
------------
Total states run:     2
Total run time:  71.898 ms in /usr/share/php/openmediavault/system/process.inc:182
Stack trace:
#0 /usr/share/php/openmediavault/engine/module/serviceabstract.inc(60): OMV\System\Process->execute()
#1 /usr/share/openmediavault/engined/rpc/config.inc(164): OMV\Engine\Module\ServiceAbstract->deploy()
#2 [internal function]: Engined\Rpc\Config->applyChanges(Array, Array)
#3 /usr/share/php/openmediavault/rpc/serviceabstract.inc(123): call_user_func_array(Array, Array)
#4 /usr/share/php/openmediavault/rpc/serviceabstract.inc(149): OMV\Rpc\ServiceAbstract->callMethod('applyChanges', Array, Array)
#5 /usr/share/php/openmediavault/rpc/serviceabstract.inc(577): OMV\Rpc\ServiceAbstract->OMV\Rpc\{closure}('/tmp/bgstatusoK...', '/tmp/bgoutputXb...')
#6 /usr/share/php/openmediavault/rpc/serviceabstract.inc(159): OMV\Rpc\ServiceAbstract->execBgProc(Object(Closure))
#7 /usr/share/openmediavault/engined/rpc/config.inc(186): OMV\Rpc\ServiceAbstract->callMethodBg('applyChanges', Array, Array)
#8 [internal function]: Engined\Rpc\Config->applyChangesBg(Array, Array)
#9 /usr/share/php/openmediavault/rpc/serviceabstract.inc(123): call_user_func_array(Array, Array)
#10 /usr/share/php/openmediavault/rpc/rpc.inc(86): OMV\Rpc\ServiceAbstract->callMethod('applyChangesBg', Array, Array)
#11 /usr/sbin/omv-engined(536): OMV\Rpc\Rpc::call('Config', 'applyChangesBg', Array, Array, 1)
#12 {main}
```

Do you have a idea for solve that ?

NB : Sorry for my bad English, I French.

Thanks you !

**Comment by luluwebmaster** [comment]
Hello,

Thanks, but how I can upgrade OpenMediaVault to the new version ?

Thank you.

**Comment by votdev** [comment]
You have to wait until openmediavault 5.0.2 is released or you have to build the package yourself if you won't wait.

**Comment by luluwebmaster** [comment]
Oh, ok thank you, I wait for the new release.

**Comment by subzero79** [comment]
@votdev also you shouldn't be flushing the filter table completely , just INPUT/OUTPUT, the panel doesn't manage the fwd chain or does it lately?

https://github.com/openmediavault/openmediavault/blob/9d50dea9cc7fa7da0201fdfd499a6943f67513a4/deb/openmediavault/usr/share/openmediavault/mkconf/iptables.d/10ifupdown#L54-L60

https://github.com/votdev/openmediavault/blob/64704884e2c62ab0ac61e6a8ba22c4e2a5a65984/deb/openmediavault/srv/salt/omv/deploy/iptables/files/etc_iptables_openmediavault-rules.j2#L18

Neither should change the policy also of the FWD chain

https://github.com/votdev/openmediavault/blob/64704884e2c62ab0ac61e6a8ba22c4e2a5a65984/deb/openmediavault/srv/salt/omv/deploy/iptables/files/etc_iptables_openmediavault-rules.j2#L71

**Comment by votdev** [comment]
> @votdev also you shouldn't be flushing the filter table completely , just INPUT/OUTPUT, the panel doesn't manage the fwd chain or does it lately?
> 
> OMV4
> 
> https://github.com/openmediavault/openmediavault/blob/9d50dea9cc7fa7da0201fdfd499a6943f67513a4/deb/openmediavault/usr/share/openmediavault/mkconf/iptables.d/10ifupdown#L54-L60
> 
> [votdev/openmediavault:deb/openmediavault/srv/salt/omv/deploy/iptables/files/etc_iptables_openmediavault-rules.j2@`6470488`#L18](https://github.com/votdev/openmediavault/blob/64704884e2c62ab0ac61e6a8ba22c4e2a5a65984/deb/openmediavault/srv/salt/omv/deploy/iptables/files/etc_iptables_openmediavault-rules.j2#L18)
> 
> Neither should change the policy also of the FWD chain
> 
> [votdev/openmediavault:deb/openmediavault/srv/salt/omv/deploy/iptables/files/etc_iptables_openmediavault-rules.j2@`6470488`#L71](https://github.com/votdev/openmediavault/blob/64704884e2c62ab0ac61e6a8ba22c4e2a5a65984/deb/openmediavault/srv/salt/omv/deploy/iptables/files/etc_iptables_openmediavault-rules.j2#L71)

Fixed, see dfa40b2959c997793f476e9783f14a2086d989fc
